### ISSUE TITLE
Failed to start the connection. SyntaxError: Unexpected token < in JSON at position 0

### ISSUE THREAD

**Issue by imsam67** [aspnet/SignalR#1536]
I'm using the latest version of SignalR that's included in ASP.NET Core 2.1 Preview-1 in my React/Redux app.

I see a similar issue posted here which was in an Angular project. Not sure what the resolution was and I seem to be experiencing the same issue.

In my app, I'm using a middleware to integrate SignalR with my Redux store.

Once I create my redux store, I call the signalRStart() function in my middleware. Here's my code in my middleware:

```
import * as types from '../../actions/action-types';
import { HubConnection } from '@aspnet/signalr';


import { setMessages } from '../../actions/messaging/messaging-actions';
import { getCurrentDomain } from '../../utils/url/url-utils';

let _store;

const currentDomain = getCurrentDomain();
const signalRHubUri = currentDomain + "/im";
let connection = new HubConnection(signalRHubUri);

export const signalRInvokeMiddleware = (store) => (next) => async (action) => {

    switch (action.type) {
        case types.SET_CHAT_ROOM_ID:
            connection.invoke("AddToGroup", action.id);
            break;
        case types.ADD_MESSAGE:
            connection.invoke("Send", action.message);
            break;
    }

    return next(action);
}

export const signalRListener = () => {

    connection.on('message', messages => {

        _store.dispatch(setMessages(messages))
    })
};


export const signalRStart = (store) => {

    _store = store;
    connection.start();
    signalRListener();
}
```
Here are the three back-to-back errors I get: 
![capture](https://user-images.githubusercontent.com/15844030/37046468-12659d70-212e-11e8-881e-4facdb353f25.PNG)

**Comment by anurse** [comment]
Can you go to the Network tab of your dev tools, capture the traces for the failing request, and then right-click in the list of requests and choose "Save as HAR with Content". Then attach the HAR file here? (Please don't paste the content inline, if you need somewhere to host it, you can use https://gist.github.com/ and paste a link)

<img width="232" alt="screen shot 2018-03-06 at 9 06 06 am" src="https://user-images.githubusercontent.com/7574/37046645-abf5834e-211d-11e8-9478-21412a6c6dc3.png">

While you're there, you can take a look at the requests SignalR initiates (`signalr.js` should be in the initiator) and see if anything looks odd (all SignalR's responses should be JSON for example, but it looks like you're getting HTML for some reason). Also, check your routing and make sure you don't have an MVC action or Razor page bound to the same URL as your Hub.

**Comment by imsam67** [comment]
As embarrassing as it is, you nailed it in the head. I was hitting my home controller instead of the hub. Because I have a SPA app, I have a catch all route in my Startup.cs but the issue was due to the fact that my SignalR declaration (which includes the route for the SignalR hub) in Configure method came after the `app.UseMvc()`.

I moved the `app.UseSignalR()` above `app.UseMvc() `and the issue disappeared.

Thank you!
